packages/x-components/src/x-modules/history-queries/store/__tests__/actions.spec.ts

```diff
@@ -369,9 +401,26 @@ describe('testing history queries module actions', () => {
     });
 
     // eslint-disable-next-line max-len
-    it('updates a history query if search response change because a filter is selected', async () => {
+    it('updates a history query when the search response changes because a filter is selected', async () => {
+      gato.totalResults = 50;
+      resetStateWith({ historyQueries: [gato, perro] });
+      await store.dispatch('updateHistoryQueriesWithSearchResponse', {
+        request: {
+          query: 'gato',
+          page: 1,
+          filters: requestFilters
+        },
+        status: 'success',
+        facets: responseFacets,
+        results,
+        totalResults
+      });
+      expectHistoryQueriesToEqual([{ ...gato, totalResults, selectedFilters }, perro]);
+    });
+
+    // eslint-disable-next-line max-len
+    it('updates a history query when the search response changes although the facet id is unknown', async () => {
       gato.totalResults = 50;
-      const selectedFilters = Object.values(requestFilters)[0];
       resetStateWith({ historyQueries: [gato, perro] });
       await store.dispatch('updateHistoryQueriesWithSearchResponse', {
```

Comment:
Testing this doesn't make much sense to me. The response shouldn't return facets with id `__unknown__`. The backend should know the id of the facet and `__unknown__` is a key used only in the front to identify a filter which we don't know the facet id.

The test I was suggesting was the behaviour of the action when a Query preview was selected. Selecting a QP creates a HQ of that request, but the QP creates the filters with facet id `__unknown__`. The test should check that the filters map correctly when the `requestFilters` have an unknown facet id.